### PR TITLE
[FSDE-1081] Grid layout attribute for Checkbox

### PR DESCRIPTION
**Pull request by riya-abraham01**

### CODE REVIEW

File: src/components/Checkbox/Checkbox.js [hunk 1]
```diff
@@ -43,6 +44,25 @@ export default class Checkbox extends React.Component {
           }
         });
 
+    Array(11).fill()
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:47]: Let's avoid the double iteration.
We need to populate `checkboxesToRender` or `checkboxesToRenderGrid`not both.
We could unify the two by using the same array `checkboxesToRender`

File: src/components/Checkbox/Checkbox.js [hunk 2]
```diff
@@ -130,4 +163,6 @@ Checkbox.defaultProps = {
   label2: 'Checkbox 2 label',
   label3: 'Checkbox 3 label',
   required: 'none',
+  inline: false,
+  grid: false,
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:167]: How do the designers control number of columns in the grid?
The difference between the Grid property and Inline is that one renders **all** the checkboxes inline and grid renders them in a controlled grid with predefined number of columns

**Review by janibolkvadze — CHANGES_REQUESTED** [review]

File: src/components/Checkbox/Checkbox.js (hunk 1 above)
Comment (riya-abraham01) [reply src/components/Checkbox/Checkbox.js:74]: double iteration removed

File: src/components/Checkbox/Checkbox.js [hunk 3]
```diff
@@ -142,6 +207,8 @@ Checkbox.propTypes = {
   deselect9: PropTypes.func,
   select10: PropTypes.func,
   deselect10: PropTypes.func,
+  /** @uxpinignoreprop */
+  inline: PropTypes.bool,
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:211]: How is this used? The idea is that all the existing prototypes using the inline still render as inline. We can not break existing prototypes

File: src/components/Checkbox/Checkbox.js [hunk 4]
```diff
@@ -32,62 +32,127 @@ export default class Checkbox extends React.Component {
...
-                    <label onClick={(e) => {
-                      toggleCheckbox(e.target, i);
-                    }} className="chi-checkbox__label" htmlFor="checkbox1">{this.props[`label${i}`]}</label>
-                  </div>
-                </div>
-            );
+      .forEach((_, i) => {
+        if (this.props[`label${i}`]) {
+          let inlineDiv;
+          if (layoutOptions !== '-inline') {
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:78]: `-inline` is used multiple times, please, define a constant in classes.js

File: src/components/Checkbox/Checkbox.js [hunk 5]
```diff
@@ -32,62 +32,127 @@ export default class Checkbox extends React.Component {
...
+            inlineDiv = <div className="chi-form__item"></div>;
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:79]: Why is this necessary?

File: src/components/Checkbox/Checkbox.js [hunk 6]
```diff
@@ -32,62 +32,127 @@ export default class Checkbox extends React.Component {
...
+          } else {
+            inlineDiv = null;
           }
-        });
+
+          checkboxesToRender.push(
+            <div className={`${layoutOptions === '-inline' ? 'chi-form__item -inline' : layoutOptions}`} key={`checkbox-${i}`}>
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:85]:
```suggestion
            <div className={`this.props.layout === 'inline' || this.props.inline ? 'chi-form__item -inline' : layoutOptions}`} key={`checkbox-${i}`}>
```
This way you don't break the existing prototypes

File: src/components/Checkbox/Checkbox.js [hunk 7]
```diff
@@ -32,62 +32,127 @@ export default class Checkbox extends React.Component {
...
+              {inlineDiv}
+              <div className="chi-checkbox">
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:87]: Can define / use constants from classes.js, please?

File: src/components/Checkbox/Checkbox.js [hunk 8]
```diff
@@ -32,62 +32,127 @@ export default class Checkbox extends React.Component {
...
+                  disabled={this.props[`disabled${i}`]}
+                  checked={this.props[`checked${i}`]}
+                  onChange={(e) => {}}
+                />
+                <label
+                  onClick={(e) => {
+                    toggleCheckbox(e.target, i);
+                  }}
+                  className="chi-checkbox__label"
+                  htmlFor="checkbox1">
```
Comment (janibolkvadze) [inline src/components/Checkbox/Checkbox.js:100]: The id is hardcoded. <input> element has no respective id

**Review by janibolkvadze — CHANGES_REQUESTED** [review]

File: src/components/Checkbox/Checkbox.js (hunk 3 above)
Comment (riya-abraham01) [reply src/components/Checkbox/Checkbox.js:214]: added changes

File: src/components/Checkbox/Checkbox.js (hunk 5 above)
Comment (riya-abraham01) [reply src/components/Checkbox/Checkbox.js:82]: removed it

**Review by janibolkvadze — APPROVED** [review]
